controller/metrics/metrics.go

```diff
@@ -44,22 +44,6 @@ type Metrics struct {
 // M contains the metrics with ControllerSubsystem as subsystem name.
 var M = New(ControllerSubsystem)
 
-// These variables are defined merely for API compatibility.
```

Comment:
are these really required? is anyone expected to use the `metrics.PersistentVolumeClaimProvisionTotal` var to `Inc()` it, without going through the `controller.metrics` instance?

Reply:
Nobody knows. These variables are part of public API of this library. And I guess everyone is afraid removing them.

I am afraid too, but let's try this:

* Please bump the library version in go.mod to `sigs.k8s.io/sig-storage-lib-external-provisioner/v10`
* Please remove `M` global variable: https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/32875745dccf6d66cca34bb48d90dd822878a4c0/controller/metrics/metrics.go#L45C5-L45C6

    * And all its usages (create a new non-global instance [here](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/082523cf04b08498bbc35fd6049800a451611f6c/controller/controller.go#L639)?)